plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPageModel.java

```diff
@@ -472,7 +472,7 @@ public void setRoute(IRoute newRoute) {
 				this.deployProject, this.projects, 
 				this.sourcePath, this.podPath, this.useInferredPodPath,
 				getService(), getServiceItems(),
-				route, this.selectDefaultRoute, this.routesByProject,
+				newRoute, this.selectDefaultRoute, this.routesByProject,
```

Comment:
>  Does it makes sense that the user can choose whatever route he wants, even a route that's not related to the service that's currently selected?

No. I don't this makes much sense from my PoV but maybe I don't have the same domain knowledge you do. I personally can't think up an example where he'd want to use one service and an unrelated route. But then I don't know all the places the routes are used aside from show in web browser / welcome page / whatever it's called now.